editors/sc-ide/widgets/help_browser.cpp

```diff
@@ -120,6 +124,41 @@ HelpBrowser::HelpBrowser( QWidget * parent ):
     setFocusProxy(mWebView);
 }
 
+void HelpBrowser::setUpServer()
+{
+
+    // setup HelpBrowser server
+    QWebSocketServer server("SCIDE HelpBrowser Server", QWebSocketServer::NonSecureMode);
+    for (int i = 0; i < 8; i++) {
+        mServerRunning = server.listen(QHostAddress::LocalHost, mServerPort);
+        if (mServerRunning) {
+            break;
+        }
+        mServerPort += 1;
+    }
+    if (mServerRunning) {
+        // setup comm channel
+        WebSocketClientWrapper clientWrapper(&server);
+        QWebChannel channel;
+        QObject::connect(&clientWrapper, &WebSocketClientWrapper::clientConnected,
+                         &channel, &QWebChannel::connectTo);
+
+        // publish IDE interface
+        IDEWebChannelWrapper ideWrapper{this};
+        channel.registerObject("IDE", &ideWrapper);
+    } else {
+        qWarning("Failed to set up help browser server.");
+    }
```

Comment:
all of these objects need to live for the lifetime of the program (or at least the help browser); best to keep this in `main` I think. you can communicate the port to the help browser with a member function (like `setServerPort(int)`)